lib/tools.js

```diff
@@ -1932,6 +1932,49 @@ function removeIdFromAllEnums(objects, id) {
     });
 }
 
+/**
+ * Validates types of obj.common properties, if invalid types are used, an error is thrown.
+ * If attributes are not provided, no error is thrown
+ *
+ * @param {object} obj an object which will be validated
+ * @throws Error if a property has the wrong type
+ */
+function validateObjectCommonProperties(obj) {
+    if (!obj || !obj.common) return;
+
+    if (obj.common.name !== undefined && typeof obj.common.name !== 'string') {
+        throw new Error(`obj.common.name has wrong type (${typeof obj.common.name}) but has to be string`);
+    }
+
+    if (obj.common.type !== undefined) {
+        if (typeof obj.common.type !== 'string') {
```

Comment:
Shouldn't we also check that one of the allowed types was passed?
https://github.com/ioBroker/ioBroker.docs/blob/master/docs/en/dev/objectsschema.md#object-types

Reply:
We can do so, I was focusing on the `common` properties.

Reply:
Oh, right... I confused that with `obj.type`. Then I guess its fine here.

But I would still add the check for obj.type, doesn't hurt.

Reply:
I made the function a bit more general and also checking the type now, due to the fact that type has to exist on every created object and has to be valid. There hasn't been a check until now.